release/models/virtual-devices/openconfig-virtual-devices.yang

```diff
@@ -0,0 +1,386 @@
+      oc-ext:telemetry-on-change;
+    }
+
+  }
+
+
+  grouping vdevice-counters-state {
+    description
+      "Operational state representing virtual device counters
+      and statistics.";
+
+    oc-ext:operational;
+
+    container counters {
+      description
+        "A collection of device-related statistics objects.";
+
+      leaf in-octets {
+        type oc-yang:counter64;
+        description
+          "The total number of octets received on the device.
+
+          In the case of storage in-octets are from the
+          request payloads for the write requests.
+
+          Sent from the compute instance.";
+      }
+
+      leaf in-pkts {
+        type oc-yang:counter64;
+        description
+          "The total number of 'packets' received on the device.
+
+          In the case of storage in-pkts are individual requests.
+
+          Sent from the compute instance.";
+      }
+
+      leaf in-discards {
+        type oc-yang:counter64;
+        description
+          "The number of inbound packets that were chosen to be
+          discarded even though no errors had been detected to
+          prevent their being deliverable to the infrastructure.";
+      }
+
+      leaf out-octets {
+        type oc-yang:counter64;
+        description
+          "The total number of octets transmitted out of the
+          virtual device, including framing characters.
+
+          In the case of storage out-octets are from the
+          completion payloads for the read requests.
+
+          Sent to the compute instance.";
+      }
+
+      leaf out-pkts {
+        type oc-yang:counter64;
+        description
+          "The total number of packets transmitted out of the
+          virtual device.
+
+          In the case of storage out-pkts are individual
+          completions.
+
+          Sent to the compute instance.";
+      }
+
+      leaf out-discards {
+        type oc-yang:counter64;
+        description
+          "The number of outbound packets that were chosen to be
+          discarded even though no errors had been detected to
+          prevent their being deliverable to the infrastructure.";
+      }
+
+      leaf last-clear {
+        type oc-types:timeticks64;
+        description
+          "Timestamp of the last time the device counters were
+          cleared.
+
+          The value is the timestamp in nanoseconds relative to
+          the Unix Epoch (Jan 1, 1970 00:00:00 UTC).";
+        oc-ext:telemetry-on-change;
+      }
+    }
+  }
+
+  grouping vdevice-limits {
+    description
+      "Reusable definition of limits of virtual device";
+
+    leaf bandwidth-limit {
+      type uint32;
+      units Mbps;
+      default 0;
+      description
+        "Bandwidth limit for virtual-device";
+    }
+
+    leaf requests-limit {
+      type uint32;
+      units pps;
+      default 0;
+      description
+        "Requests limit for virtual-device";
+    }
+  }
+
+  grouping vdevice-host-virtual-ref {
```

Comment:
At some point, we will need to accomodate hstid:pfid:vfid:ftype.
We will need to clarify what the format of the strings should be.
OK with committing for now.